pages/dashboard.tsx

```diff
@@ -66,7 +68,10 @@ const Dashboard: NextPage = () => {
   const [protocolList, setProtocolList] = useState([]);
   const [protocolLink, setProtocolLink] = useState("");
   const [linkModalOpen, setLinkModalOpen] = useState(false);
-  const [subscriptionStatus, setSubscriptionStatus] = useState<Boolean>(false);
+  const dispatch = useDispatch();
```

Comment:
We don't need to dispatch the value in every react component, just dispatch once and then use selector to fetch everywhere

Reply:
We need the `walletAddress` to fetch the `subscriptionStatus`. If a user browses the home page and connects their wallet, then browses the dashboard page and switches wallets, we won't be able to determine the new `walletAddress`'s `subscriptionStatus`.
Does it make sense?

Reply:
Oh good point! But what do you think if you do some research about the best practice for this scenario?

Since it might be challenging to make the redux states consistent between different pages.

Since we need to remember always manually dispatch the latest data with the latest wallet. It might not be feasible once we have more and more redux slice🥲